src/main/java/com/example/demo/webclient/openfeignclient/OpenWeatherApi.java

```diff
@@ -0,0 +1,33 @@
+package com.example.demo.webclient.openfeignclient;
+
+import com.example.demo.webclient.Forecast;
+import com.example.demo.webclient.Location;
+import feign.Param;
+import feign.RequestLine;
+
+import java.util.List;
+
+public interface OpenWeatherApi {
+
+    @RequestLine("GET " +
+            "?" +
+            "lon" + "=" + "{longitude}" + "&" +
+            "lat" + "=" + "{latitude}" + "&" +
+            "units" + "=" + "{units}" + "&" +
+            "apiKey" + "=" + "{apiKey}"
+    )
+    Forecast getForecast(@Param String latitude,
+                         @Param String longitude,
+                         @Param String units,
+                         @Param String apiKey);
```

Comment:
```suggestion
@GetMapping
    Forecast getForecast(
            @RequestParam("lon") String longitude,
            @RequestParam("lat") String latitude,
            @RequestParam String units,
            @RequestParam String apiKey);
```
В мене працює без того, щоб вручну конструюювати запит. Власне ми для цього і хочемо мати клієнт, щоб не склеювати стрінги вручну.

Reply:
Закомітив як у Вас і тести перестали працювати. Кидає багато помилок. Як тільки прописую запит вручну, то все знову починає працювати

Reply:
В основному кидає, щось типу такого Caused by: org.springframework.beans.BeanInstantiationException: Failed to instantiate [com.example.demo.webclient.openfeignclient.OpenWeatherApi]: Factory method 'getFeignClientForWeatherForecast' threw exception with message: Method OpenWeatherApi#getForecast(String,String,String,String) not annotated with HTTP method type (ex. GET, POST)

Reply:
Це я трохи Вас заплутав. Якщо використовувати feign без Spring, то дійсно, доведеться конструювати запит вручну. Якщо перейти на 
`implementation 'org.springframework.cloud:spring-cloud-starter-feign:1.4.7.RELEASE'`
то буде простіше, можна використовувати анотації Spring. Приклади тут: https://docs.spring.io/spring-cloud-openfeign/docs/current/reference/html/